apps/cyberstorm-remix/app/root.tsx

```diff
@@ -142,13 +143,28 @@ export async function clientLoader() {
     publicEnvVariables.VITE_COOKIE_DOMAIN
   );
 
-  // We need to run this here too in addition to the, shouldRevalidate function,
-  // as for some reason the commtis to localStorage are not done before the the clientLoader is run
-  sessionTools.sessionValid(
-    publicEnvVariables.VITE_API_URL,
-    publicEnvVariables.VITE_COOKIE_DOMAIN
+  let forceUpdateCurrentUser = false;
+  if (
+    request.url.startsWith(`${publicEnvVariables.VITE_BETA_SITE_URL}/teams`) ||
+    request.url.startsWith(`${publicEnvVariables.VITE_BETA_SITE_URL}/settings`)
+  ) {
+    forceUpdateCurrentUser = true;
+  } else {
+    // In all other cases check if actually need to fetch
+    // current-user data. Ideally we shouldn't need to do
+    // this sessionValid check again, but for some reason
+    // we need to run this here too in addition to the,
+    // shouldRevalidate function, cause for some reason
+    // the commits to localStorage are not done before
+    // the clientLoader is run.
+    sessionTools.sessionValid(
```

Comment:
I feel this is bad pattern. Based on the naming I'd assume the function return a boolean. Well, it does, but the value is ignored. So why did we call it? Because the function had some side effects that are completely unknown to me until I read that function.

~~I think this should be split to two functions, isSessionValid that returns a boolean without having side effects, and another with a descriptive name for the side effects it has.~~ Scratch that, looks the boolean value is never used anywhere, so we can drop the return value and just have function that updates the staleness. (Assuming this is still needed after the earlier comment is resolved.)